File: 240707/괄호 쌍의 개수/number-of-parenthesis-pairs.py
tmp = input()
tmp = list(tmp)
def is_correct(tmp):
    a,b = 0,0
    for t in tmp:
        if t =='(':
            a +=1
        else:
            b+=1
    return a ==b

cnt =0

for i in range(len(tmp)): #하나씩 변경
    if tmp[i] == '(':
        tmp[i] = ')'
    else:
        tmp[i] = '('
    #print(tmp)
    if not is_correct(tmp):
        if tmp[i] == '(': tmp[i] =')'
        else: tmp[i] ='('
        continue
    a,b = 0,0
    success = True
    for j in range(len(tmp)):
        if tmp[j] == '(':
            a +=1
        else:
            b +=1
        if a <b:
            success = False
            break
    if success:
        cnt +=1
    if tmp[i] == '(': tmp[i] =')'
    else: tmp[i] = '('
print(cnt)

File: 240707/괄호 쌍의 개수/number-of-parenthesis-pairs.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

bool is_balanced_and_valid(const string &s) {
    int balance = 0;
    for (char c : s) {
        if (c == '(') {
            balance++;
        } else {
            balance--;
        }
        if (balance < 0) {
            return false;
        }
    }
    return balance == 0;
}

int main() {
    string tmp;
    cin >> tmp;

    int cnt = 0;
    for (size_t i = 0; i < tmp.size(); ++i) {
        char original = tmp[i];
        tmp[i] = (tmp[i] == '(') ? ')' : '(';  // Flip the bracket
        if (is_balanced_and_valid(tmp)) {
            cnt++;
        }
        tmp[i] = original;  // Restore the original bracket
    }

    cout << cnt << endl;
    return 0;
}